### PR TITLE
fix(DTFS2-7021): Ensure EC with draft mode set to `false` are used

### PR DESCRIPTION
**Pull request by abhi-markan**
## Introduction :pencil2:
Ensure eligibility criterions for any production should refer to the latest version alongside with EC which are not in draft mode.

## Resolution :heavy_check_mark:
* Added `isInDraft: false` to `BSS/EWCS` find compound condition.
* Added `$eq` operator for `GEF` find call.

## Miscellaneous ➕ 
* Formatting fixes
* Added tests

### CODE REVIEW

**Review by avaitonis — APPROVED** [review]
LGTM, but could have tests

**Review by ChristianMcc — DISMISSED** [review]

**Review by ChristianMcc — DISMISSED** [review]

File: portal-api/src/v1/controllers/eligibilityCriteria.controller.js [hunk 1]
```diff
@@ -50,14 +50,27 @@ exports.findAll = (req, res) =>
...
+    ),
+  );
 
 exports.findOne = (req, res) => findOneEligibilityCriteria(Number(req.params.version), (eligibilityCriteria) => res.status(200).send(eligibilityCriteria));
 
+/**
+ * Finds the latest eligibility criteria document from the 'eligibilityCriteria' collection.
+ * EC is returned as an array for mapping.
+ * 
+ * @returns {Promise<Object>} The latest eligibility criteria document.
```
Comment (ChristianMcc) [inline portal-api/src/v1/controllers/eligibilityCriteria.controller.js:63]: Is this Promise<Object[]>?

**Review by ChristianMcc — CHANGES_REQUESTED** [review]
Had a closer look and suggested some naming/documentation updates. All implementation is good just improving readability hopefully!

File: portal-api/src/v1/controllers/eligibilityCriteria.api-test.js [hunk 2]
```diff
@@ -3,52 +3,58 @@ const db = require('../../drivers/db-client');
...
-      { id: 1, version: 1, isInDraft: false },
-      { id: 2, version: 2, isInDraft: true },
-    ]),
-  };
-  const mockDb = {
-    getCollection: jest.fn().mockResolvedValueOnce(mockCollection),
-  };
-
-  db.getCollection = mockDb.getCollection;
+it('should return the latest eligibility criteria object for BSS_EWCS deal type when the latest EC is in draft mode', async () => {
```
[the remark this answers is not in the document]
Comment (abhi-markan) [reply portal-api/src/v1/controllers/eligibilityCriteria.api-test.js:43]: Updated.

File: portal-api/src/v1/controllers/eligibilityCriteria.controller.js [hunk 3]
```diff
@@ -57,25 +57,20 @@ exports.findAll = (req, res) =>
...
- *
- * @param {Object} params - The parameters for creating the number.
- * @param {string} params.dealType - The type of the deal.
- * @param {string} params.entityType - The type of the entity.
- * @param {string} params.entityId - The ID of the entity.
- * @param {string} params.dealId - The ID of the deal.
- * @param {string} params.user - The user associated with the deal.
- * @returns {Promise<Object>} The generated number.
- * @throws {Error} If there is an error getting the number for the deal.
+ * Finds the latest eligibility criteria document from the 'eligibilityCriteria' collection.
```
[the remark this answers is not in the document]
Comment (abhi-markan) [reply portal-api/src/v1/controllers/eligibilityCriteria.controller.js:60]: Good shout, have updated.

File: portal-api/src/v1/controllers/eligibilityCriteria.controller.js [hunk 4]
```diff
@@ -57,25 +57,20 @@ exports.findAll = (req, res) =>
...
+ * EC is returned as an array for mapping.
+ * 
+ * @returns {Promise<Object>} The latest eligibility criteria document.
  */
 const findLatest = async () => {
```
[the remark this answers is not in the document]
Comment (abhi-markan) [reply portal-api/src/v1/controllers/eligibilityCriteria.controller.js:65]: Have updated.

File: portal-api/src/v1/controllers/eligibilityCriteria.api-test.js [hunk 5]
```diff
@@ -0,0 +1,60 @@
+// Returns the latest eligibility criteria object for BSS_EWCS deal type when there is only one object in the collection
```
[the remark this answers is not in the document]
Comment (abhi-markan) [reply portal-api/src/v1/controllers/eligibilityCriteria.api-test.js:1]: Have removed.

File: portal-api/src/v1/controllers/eligibilityCriteria.controller.js (hunk 1 above)
Comment (abhi-markan) [reply portal-api/src/v1/controllers/eligibilityCriteria.controller.js:63]: Good spot, nope just an object have updated the DocString for both.

**Review by ChristianMcc — APPROVED** [review]
LGTM :) tests are failing so imagine there's something crashing central API that can't compile right now

Will re-approve once you've fixed

**Review by ChristianMcc — APPROVED** [review]

**Review by ChristianMcc — APPROVED** [review]

**Pull request merged**